docker-compose.yml

```diff
@@ -0,0 +1,100 @@
+# Docker compose from bringing up a local KSQL cluster and dependencies.
+#
+# You can scale the number of KSQL nodes in the cluster by using the docker `--scale` command line arg.
+# The default is one primary node and 1 secondary node. The only difference is that the primary node
+# has a well-known port exposed so clients can connect, where as the secondary nodes use auto-port
+# assignment so that ports don't clash.  You can scale out the secondary nodes, e.g. to start a three node cluster run:
+# > docker-compose up --scale ksql-server=2
+#
+# If you wish to run with locally built KSQL docker images then build then:
+#
+# 1. ensure logged in to docker:
+# > docker login
+#
+# 2. ensure you can log in to AWS ECR:
+# > aws ecr get-login --no-include-email | sh
+#
+# 3. build docker images from local changes (Note: access to Confluent docker registry is required):
+# Change `docker.upstream-tag` if you want to depend on anything other than the latest master upstream, e.g. 5.4.x-latest
+# > mvn -Pdocker package -DskipTests -Dspotbugs.skip -Dcheckstyle.skip  -Ddockerfile.skip=false -Dskip.docker.build=false -Ddocker.upstream-tag=master-latest -Ddocker.tag=local.build -Ddocker.upstream-registry='368821881613.dkr.ecr.us-west-2.amazonaws.com/'
+#
+# 4. check images build:
+# > docker image ls | grep ksql.local.build
+# You should see your new images listed
+#
+# 5. update this file below replacing all references to "confluentinc/ksqldb-server:latest" with your image, e.g. "placeholder/confluentinc/ksql-rest-app:local.build"
+
+---
+version: '2'
+services:
+  zookeeper:
+    image: confluentinc/cp-zookeeper:latest
+    environment:
+      ZOOKEEPER_CLIENT_PORT: 32181
+      ZOOKEEPER_TICK_TIME: 2000
+
+  kafka:
+    image: confluentinc/cp-enterprise-kafka:latest
+    ports:
+      - "29092:29092"
+    depends_on:
+      - zookeeper
+    environment:
+      KAFKA_BROKER_ID: 1
+      KAFKA_ZOOKEEPER_CONNECT: zookeeper:32181
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://kafka:9092,PLAINTEXT_HOST://localhost:29092
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
+      KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR: 1
+      KAFKA_TRANSACTION_STATE_LOG_MIN_ISR: 1
+      KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS: 100
+
+  schema-registry:
+    image: confluentinc/cp-schema-registry:latest
+    depends_on:
+      - zookeeper
+      - kafka
+    environment:
+      SCHEMA_REGISTRY_HOST_NAME: schema-registry
+      SCHEMA_REGISTRY_KAFKASTORE_CONNECTION_URL: zookeeper:32181
+
+  ksqldb-server-primary:
+    image: confluentinc/ksqldb-server:latest
+    hostname: ksqldb-server-primary
+    container_name: ksqldb-server-primary
+    depends_on:
+      - kafka
+      - schema-registry
+    ports:
+      - "8088:8088"
+    environment:
+      KSQL_LISTENERS: http://0.0.0.0:8088
+      KSQL_BOOTSTRAP_SERVERS: kafka:9092
```

Comment:
Just want to check if this should be 29092? since thats the port that gets exposed for the kafka container?

Reply:
This is correct.